src/map/script.c

```diff
@@ -9809,6 +9809,23 @@ static BUILDIN(statusup2)
 	return true;
 }
 
+
+/*==========================================
+* Returns the number of stat points needed to change the specified stat by val.
+* needed_status_point(<type>,<val>{,<char id>}); [secretdataz]
+*------------------------------------------*/
+static BUILDIN(needed_status_point)
+{
+	int type = script_getnum(st, 2);
+	int val = script_getnum(st, 3);;
+	struct map_session_data *sd = script->rid2sd(st);
+
+	if (sd != NULL)
```

Comment:
will be script error if sd not attached.
probably better return some error status in this case?

Reply:
wait, aren't the `script->rid2sd(st)` would throw the error status if `sd == NULL`?

Reply:
i mean bit other. yes error in server console is fine
but current code also terminate npc script, because here no pushint in else branch.
if you add pushint also on errors, npc script can get result and continue running

Reply:
ops, rid2sd terminating script already. then probably better replace rid2sd to map->id2sd(st->rid)

Reply:
i think its better to remain `script->rid2sd(st);`.
the script command are designed for player to retrieve that status points. 
Its fine to terminate the script if no player are attached to it, just like any other script commands like `*statusup()` etc